### PR TITLE
add new request to data source

### PR DESCRIPTION
**Pull request by ChengaDev**

### CODE REVIEW

File: Admin/Api/Api.ts [hunk 1]
```diff
@@ -57,6 +60,91 @@ class AdminApi {
...
+            .send('Request object not found')
+            .end();
+        }
+        res
+          .header('Content-Type', 'application/json')
+          .send(request)
+          .end();
+      } catch (e) {
+        console.log(e);
+        res.send({ error: e, success: false }).end();
```
Comment (eladbash) [inline Admin/Api/Api.ts]: why do we need the success?

File: Core/Repositories/Adapters/RequestFileSystemRepository.ts [hunk 2]
```diff
@@ -4,13 +4,82 @@ import { Request } from '../../Models/request';
 
 class RequestRepository implements Port.IRequestRepository {
   private _requestsConfigFilePath: string =
```
Comment (eladbash) [inline Core/Repositories/Adapters/RequestFileSystemRepository.ts]: Doesn't related to this review but I would rename ```private _requestsConfigFilePath to private readonly REQUESTS_FILE_PATH``` - The capital letters indicates a const.

File: Admin/Api/Api.ts (hunk 1 above)
Comment (ChengaDev) [reply Admin/Api/Api.ts]: removed

File: Core/Repositories/Adapters/RequestFileSystemRepository.ts (hunk 2 above)
Comment (ChengaDev) [reply Core/Repositories/Adapters/RequestFileSystemRepository.ts]: Agree, done

File: Admin/Api/Api.ts [hunk 3]
```diff
@@ -57,6 +60,118 @@ class AdminApi {
...
+     */
+    app.put('/request/:id', (req: express.Request, res: express.Response) => {
+      const id = req.params.id;
+      const name = req.body.name;
+      const relativePath = req.body.relativePath;
+
+      // validate all inputs supplied
+      if (!name || !relativePath) {
+        res.statusCode = 500;
+        res.send({ error: 'Sorry, Missing parameter', success: false }).end();
```
Comment (eladbash) [inline Admin/Api/Api.ts]: The same about "success"

File: Admin/Api/Api.ts [hunk 4]
```diff
@@ -57,6 +60,118 @@ class AdminApi {
...
+      }
+
+      const request = new Request(id, relativePath, name);
+      try {
+        this._requestRepository.update(request);
+        res.send({ error: '', success: true }).end();
```
Comment (eladbash) [inline Admin/Api/Api.ts]: the same - see if there are more places

File: Admin/Api/Api.ts [hunk 5]
```diff
@@ -57,6 +60,118 @@ class AdminApi {
...
+     */
+    app.delete(
+      '/request/:id',
+      (req: express.Request, res: express.Response) => {
+        const id = req.params.id;
+
+        if (!id) {
+          res
+            .status(500)
+            .send({ error: 'Sorry, Missing parameter: id', success: false })
```
Comment (eladbash) [inline Admin/Api/Api.ts]: here

File: Admin/Api/Api.ts [hunk 6]
```diff
@@ -57,6 +60,118 @@ class AdminApi {
...
+            .end();
+        }
+
+        try {
+          this._requestRepository.remove(id);
+          res.send({ error: '', success: true }).end();
```
Comment (eladbash) [inline Admin/Api/Api.ts]: here

**Pull request merged**